nrfx/hal/nrf_clock.h

```diff
@@ -772,6 +772,17 @@ NRF_STATIC_INLINE bool nrf_clock_start_task_check(NRF_CLOCK_Type const * p_reg,
     }
 }
 
+/*
+ * `-Warray-bounds` warning is disabled for the `nrf_clock_is_running`
+ * function because GCC 12 and above may report a false positive due to the
```

Comment:
if this only affects GCC 12.x, should we also check for version? https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

Reply:
There is no guarantee that the commit that introduced this issue in the GCC 12 will not be backported to older releases (e.g. to GCC 10.x and GCC 11.x), also the future releases may still be affected by this issue.

I would not make it any more restrictive than it is necessary.